services/catalog/src/simcore_service_catalog/api/routes/services_specifications.py

```diff
@@ -0,0 +1,68 @@
+import logging
+
+from fastapi import APIRouter, Depends, HTTPException, Query, status
+from models_library.services import ServiceKey, ServiceVersion
+from models_library.users import UserID
+
+from ...db.repositories.groups import GroupsRepository
+from ...db.repositories.services import ServicesRepository
+from ...models.schemas.constants import RESPONSE_MODEL_POLICY
+from ...models.schemas.services_specifications import (
+    ServiceSpecifications,
+    ServiceSpecificationsGet,
+)
+from ...services.function_services import is_function_service
+from ..dependencies.database import get_repository
+from ..dependencies.services import get_default_service_specifications
+
+router = APIRouter()
+logger = logging.getLogger(__name__)
+
+
+@router.get(
+    "/{service_key:path}/{service_version}/specifications",
+    response_model=ServiceSpecificationsGet,
+    **RESPONSE_MODEL_POLICY,
+)
+async def get_service_specifications(
+    user_id: UserID,
+    service_key: ServiceKey,
+    service_version: ServiceVersion,
+    strict: bool = Query(
+        False,
+        description="if True only the version specs will be retrieved, if False the latest version will be used instead",
+    ),
```

Comment:
I did not understand this part quite well. I seem not to grasp the use case. Event after looking at the tests.

Reply:
so in case strict is True, only the specifications of the version asked for will be returned, or default if there are none.
if strict is False, then the "latest" specifications for that user will be returned (e.g. jupyter-math:2.5.0 will get the specs from jupyter-math:2.4.2 if they exist) or default if there are none.